packages/api/src/share_link/validators.ts

```diff
@@ -49,3 +49,59 @@ export function validateCreateShareLinkRequest(
     throw validation.error;
   }
 }
+
+export function validateUpdateShareLinkRequest(
+  data: unknown
+): asserts data is UpdateShareLinkRequest {
+  const validation = Joi.object()
+    .keys({
+      ...fieldsFromUserAuthentication,
+      permissionsLevel: Joi.string()
+        .when("accessRestrictions", {
+          is: "none",
+          then: Joi.valid("viewer"),
+          otherwise: Joi.valid(
+            "viewer",
```

Comment:
FWIW if you do go the enum route you would be able to do something like `Object.values(PermissionTypes)` here and avoid the repetition.